src/Microsoft.IdentityModel.Tokens/Validation/ValidationParameters.cs

```diff
@@ -440,16 +440,16 @@ public string RoleClaimType
         public TokenReplayValidator TokenReplayValidator { get; set; }
 
         /// <summary>
-        /// Gets or sets a delegate that will be used to validate the type of the token.
-        /// If the token type cannot be validated, an exception MUST be thrown by the delegate.
+        /// Allows overriding the delegate that will be used to validate the type of the token.
+        /// If the token type cannot be validated, a <see cref="TokenTypeValidationResult"/> MUST be returned by the delegate.
         /// Note: the 'type' parameter may be null if it couldn't be extracted from its usual location.
         /// Implementations that need to resolve it from a different location can use the 'token' parameter.
         /// </summary>
         /// <remarks>
-        /// If set, this delegate will be called to validate the 'type' of the token, instead of default processing.
-        /// This means that no default 'type' validation will occur.
+        /// If no delegate is set, the default implementation will be used. The default checks the type
+        /// against the <see cref="ValidTypes"/> property, if it's present then, it will succeed.
         /// </remarks>
-        public TypeValidator TypeValidator { get; set; }
+        public TypeValidatorDelegate TypeValidator { get; set; } = Validators.ValidateTokenType;
```

Comment:
We should not allow this to be set to null.

Reply:
How's the approach below? although it is somewhat of an anti-pattern to throw an exception. Let's discuss this offline. 

private TypeValidatorDelegate _typeValidator = Validators.ValidateTokenType;

public TypeValidatorDelegate TypeValidator
{
    get { return _typeValidator; }
    set
    {
        if (value == null)
        {
            throw new ArgumentNullException(nameof(value), "TypeValidator cannot be set as null.");
        }
        _typeValidator = value;
    }
}

Reply:
yes, we sometimes write this as
```cs
_typeValidator = value ?? throw new ArgumentNullException(nameof(value), "TypeValidator cannot be set as null.");
```

Reply:
We should have a single test showing that setting null faults.

Reply:
Added test on separate file and updated code to prevent null setting of delegates. Thanks!